src/charts/heatmap.js

```diff
@@ -0,0 +1,350 @@
+
+            container
+                .append('g').classed('chart-group', true);
+            container
+                .append('g').classed('day-labels-group', true);
+            container
+                .append('g').classed('hour-labels-group', true);
+            container
+                .append('g').classed('metadata-group', true);
+        }
+
+        /**
+         * Cleaning data casting the values to the proper
+         * type while keeping the rest of properties on the data. It
+         * also creates a set of zeroed data (for animation purposes)
+         * @param   {HeatmapData} originalData  Raw data as passed to the container
+         * @return  {HeatmapData}               Clean data
+         * @private
+         */
+        function cleanData(originalData) {
+            return originalData.reduce((acc, {day, hour, value}) => {
+                return [
+                    ...acc,
+                    {
+                        day: +day,
+                        hour: +hour,
+                        value: +value
+                    }
+                ];
+            }, []);
+        }
+
+        /**
+         * Creates the scales for the heatmap chart
+         * @return void
+         */
+        function buildScales() {
+            colorScale = d3Scale.scaleLinear()
+                .range([colorSchema[0], colorSchema[colorSchema.length - 1]])
+                .domain(d3Array.extent(data, getValue))
+                .interpolate(d3Interpolate.interpolateHcl);
+        }
+
+        /**
+         * Draws the boxes of the heatmap
+         */
+        function drawBoxes() {
+            boxes = svg.select('.chart-group').selectAll('.box').data(data);
+
+            boxes.enter()
+              .append('rect')
+                .classed('box', true)
+                .attr('width', boxSize)
+                .attr('height', boxSize)
+                .attr('x', ({hour}) => hour * boxSize)
+                .attr('y', ({day}) => day * boxSize)
+                .style('opacity', boxInitialOpacity)
+                .style('fill', boxInitialColor)
+                .style('stroke', boxBorderColor)
+                .style('stroke-width', boxBorderSize)
+                .transition()
+                    .duration(animationDuration)
+                    .style('fill', ({value}) => colorScale(value))
+                    .style('opacity', boxFinalOpacity);
+
+            boxes.exit().remove();
+        }
+
+        /**
+         * Draws the day labels
+         */
+        function drawDayLabels() {
+            let dayLabelsGroup = svg.select('.day-labels-group');
+
+            dayLabels = svg.select('.day-labels-group').selectAll('.day-label')
+                .data(daysHuman);
+
+            dayLabels.enter()
+              .append('text')
+                .text((label) => label)
+                .attr('x', 0)
+                .attr('y', (d, i) => i * boxSize)
```

Comment:
I see that there is multiple instances where `x * boxSize` where `x` is anything. Could be useful to create a private function `withBoxSize = (x) => x * boxSize` and then use like `.attr('y', (d, i) => withBoxSize(i))`. Doesn't have to be that way though, just a thought.

Reply:
Yeah, that would make sense if we keep the functionality as it is right now.

Reply:
However, I think that with the next steps, the boxes will stop being squares and start being rectangles.